sound/soc/sof/pm.c

```diff
@@ -147,6 +147,13 @@ static int sof_restore_pipelines(struct snd_sof_dev *sdev)
 		struct sof_ipc_reply reply;
 		struct sof_ipc_dai_config *config = dai->dai_config;
 
+		/*
+		 * Invalidate link dma channel upon resume.
+		 * This will be assigned during hw_params.
```

Comment:
why can't we keep the same assignment on resume?

Reply:
@plbossart because the link DMA channel will change when the link hw_params is done after resume. So its better to remove the stale channel info from the DAI config.